util/reggen/reg_block.py

```diff
@@ -153,6 +163,55 @@ def add_raw(self, where: str, raw: object) -> None:
 
         handlers[entry_type](entry_where, entry_body)
 
+    def _validate_async(self, name: Optional[str]) -> None:
```

Comment:
I think this check could be moved into `add_raw`, where we can check the register being added? That gets rid of the validate method that needs calling afterwards, which is a bit nicer. I think it also makes the code a bit simpler because we don't have to iterate when checking.